### PR TITLE
Exposure : enable color matching between pictures

### PR DESCRIPTION
**Pull request by aurelienpierre**
This is the companion of #10579 for lightness mapping. It works exactly the same, but at the exposure module level.

TL; DR : you can choose to remap any region of the picture to an arbitrary lightness.

Also, this removes the previous color-picker auto-tune that was based on min/max RGB detection, suitable only for display-referred workflow and aiming at avoiding clipping in this context.

Screenshot_20211207_113046

### CODE REVIEW

File: src/iop/exposure.c [hunk 1]
```diff
@@ -829,20 +1052,79 @@ void gui_init(struct dt_iop_module_t *self)
...
 
-  g->autoexpp = dt_color_picker_new(self, DT_COLOR_PICKER_AREA,
-                dt_bauhaus_slider_new_with_range(self, 0.0, 0.2, .001, 0.01, 3));
-  gtk_widget_set_tooltip_text(g->autoexpp, _("percentage of bright values clipped out, toggle color picker to activate"));
-  dt_bauhaus_slider_set_format(g->autoexpp, "%.3f%%");
-  dt_bauhaus_widget_set_label(g->autoexpp, NULL, N_("clipping threshold"));
-  g_signal_connect(G_OBJECT(g->autoexpp), "value-changed", G_CALLBACK(autoexpp_callback), self);
-  gtk_box_pack_start(GTK_BOX(vbox_manual), GTK_WIDGET(g->autoexpp), TRUE, TRUE, 0);
+  g->spot_settings = dt_bauhaus_combobox_new(self);
+  dt_bauhaus_widget_set_label(g->spot_settings, NULL, N_("spot exposure mapping settings"));
```
Comment (elstoc) [inline src/iop/exposure.c:1064]: As in the other PR, IMO the word "settings" is superfluous in this label.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Somehow same comments as for the color calibration changes:
1. "settings" is superfluous in label "spot exposure mapping settings"
2. I would probably move this to the bottom of the module
3. In the tooltip put "correction" & "mesure" inside quotes

**Comment by TurboGit** [comment]
Now another question, how do you see the two modes "automatic" and this PR? Aren't they redundant?

**Comment by aurelienpierre** [comment]
> Now another question, how do you see the two modes "automatic" and this PR? Aren't they redundant?

Automatic is automatic. It performs histogram normalization based on the distribution of intensities. That may be good for exposure normalization of time-lapses with a fixed camera on a tripod and doesn't require user input after copy-pasting the history on a batch of pictures. 

But that's too blind to the actual image content to work in all cases, for example if the camera is moving and the ratio of highlights vs. shadows changes in the background, the histogram distribution will not allow to anchor the exposure for the subject because it has no idea where the subject is.

So this is a more supervised way of anchoring the exposure over a particular surface of a particular object without the intermediate assumptions over histogram distribution.

**Comment by elstoc** [comment]
As with the color matching PR, I think this needs a default behaviour for when people aren't performing exposure mapping (the most common use case). Three options

- Have the existing color picker default to 50% lightness when the spot exposure mapping section is collapsed
- Move the color picker into the collapsible section (to the lightness slider)
- Have a second color picker in the collapsible section that targets the measured lightness, and keep the existing color picker to always correct to 50% lightness

As with the color matching PR, I prefer the last option (even though the two pickers might be considered redundant).

The tooltip on the collapsible section seems to have been copied-and-pasted from the color matching PR. Similarly the spot mode tooltip references "color" which looks wrong to me.

**Comment by elstoc** [comment]
I think this probably also resolves #6588

**Comment by TurboGit** [comment]
@aurelienpierre : I'd like to make this moving. I'll update this PR to use the new collapsible section support. I'll create a new PR with this. I'll test and if you are OK I'll merge then. I understand the different point of view about the color-picker. We'll revisit this if needed after more feedback. On my side after testing it seems ok as-is.

**Comment by TurboGit** [comment]
@aurelienpierre : Closing, see #11142

**Comment by estaf1001** [comment]
Unfortunately, in DT 3.8.1 there is no such thing as spot mode in the exposure tool.
<a href="https://imgbb.com/"></a>

**Comment by elstoc** [comment]
Look at the milestone on the PR. It's scheduled for v4.0.

**Pull request closed without merging**